apps/explorer/src/components/orders/OrdersUserDetailsTable/index.tsx

```diff
@@ -118,6 +161,14 @@ const RowOrder: React.FC<RowProps> = ({ order, isPriceInverted }) => {
 const OrdersUserDetailsTable: React.FC<Props> = (props) => {
   const { orders, messageWhenEmpty } = props
   const [isPriceInverted, setIsPriceInverted] = useState(false)
+  const [showCanceledAndExpired, setShowCanceledAndExpired] = useState(false)
+  const [showPreSigning, setShowPreSigning] = useState(false)
+
+  const canceledAndExpiredCount = orders ? orders.filter(isExpiredOrCanceled).length : 0
+  const preSigningCount = orders ? orders.filter((order) => order.status === 'signing').length : 0
+  const showFilter = canceledAndExpiredCount > 0 || preSigningCount > 0
+  const allOrdersAreHidden =
```

Comment:
Nitpick: I feel like it makes more sense to phrase it as a question than an affirmation. But feel free to ignore.

```suggestion
  const areOrdersAllHidden =
```